### ISSUE TITLE
Patches for Version 5

### ISSUE THREAD

**Issue by davrot** [smhaller/ldap-overleaf-sl#50]
I created diffs that can be applied to version 5: 
https://github.com/davrot/ldap-overleaf-sl

However, I couldn't / will not test them because I am too stupid to get an Authentik server running and was not even able to run a test OAuth2 in Python against it. Thus I will give up on the Authentik server idea and will modify the overleaf password part to be connected directly to a real LDAP server. 

Long story short, maybe the diffs and scripts, I did last night, are helpful for somebody...

**Comment by yzx9** [comment]
Thanks! However, I'll need to test it next weekend. Any reviews in the meantime would be greatly appreciated.

**Comment by VPaulV** [comment]
I'd be happy to test it. Could you provide instructions on what needs to be done? Would simply applying diffs be sufficient, or should something else be changed in my docker-compose/config as well?

P.S: sorry for the off-topic. @davrot - try setting up authentik+traefik: https://goneuland.de/authentik-in-kombination-mit-traefik/

**Comment by davrot** [comment]
> VPaulV

Maybe I summarize what I did with https://github.com/davrot/ldap-overleaf-sl:

1. I modified scripts/make_diffs.sh: "diff -u " Then I downloaded the old files and applied the old patches. After that I re-created the old patches with more context than just the line numbers. 

2. I hacked together ldap-overleaf-sl/extract_files_from_git.sh which takes the required files from the overleaf git.

2b. I made a diff between the old unpatched filed and the new unpatched files, to understand the changes done since then. 

3. ldap-overleaf-sl/apply_diffs_5.sh applies the new new patches to the new files. Most of the re-created old patches with the -u extended information worked out of the box. Two or three patches, I had to modify according the information from 2b.


@VPaulV At that point I noticed that I am too stupid for authentik. The installation seemed to work fine. I made even my own ansible installation script:

https://github.com/davrot/overleaf_ansible/blob/main/useless/install_authentik.yaml

But I was not able to connect my portainer ( https://docs.goauthentik.io/integrations/services/portainer/ ) to it nor to get a valid token with python. I tried stuff like this (and variants)

client = BackendApplicationClient(client_id=client_id)
oauth = OAuth2Session(client=client)
token = oauth.fetch_token(
    token_url=token_url,
    client_id=client_id,
    client_secret=client_secret,
    include_client_id=True,
    verify=False,
)

for hours. At that point I was pondering my life decision. Maybe I should do something with wood or garden?

**Comment by davrot** [comment]
oh, BTW there might be a problem with the update of Tex. The Tex people moved to 2024 and the overleaf docker has still 2023.

I worked around it by creating a volume (please ignore the ansible style :-) ):

    - name: Create a volume overleaf_tex2024
      community.docker.docker_volume:
        name: overleaf_tex2024
        state: present

and use the docker volume function to put it at the position of 2023:

        volumes:
            - "overleaf_data:/var/lib/overleaf"
            - "overleaf_tex2024:/usr/local/texlive/2024"
            - "overleaf_tex2024:/usr/local/texlive/2023"
           

And finally I auto installed tex 2024 into the volume:

---
- name: update tex
  hosts: all
  become: true

  tasks:

    - name: Update tex
      community.docker.docker_container_exec:
        container: overleaf
        command: /bin/bash -c "cd /root ; rm -rf install-tl-* ;  curl -L -o tl-unx.tar.gz https://mirror.ctan.org/systems/texlive/tlnet/install-tl-unx.tar.gz ; tar -xzf tl-unx.tar.gz; cd install-tl-*; perl ./install-tl --no-interaction "

**Comment by thmo** [comment]
Is there by any chance an updated patch set for the current version (5.1.1) somewhere?

**Comment by armanddidierjean** [comment]
Hi @thmo, if you only need OAuth 2 and don't use LDAP, you may use https://github.com/aeecleclair/ldap-overleaf-sl/tree/5.1.1
Don't forget to update your docker-compose and migrate your database if you want to upgrade an existing installation
